influxql/ast.go

```diff
@@ -2698,6 +2653,47 @@ func CloneExpr(expr Expr) Expr {
 	panic("unreachable")
 }
 
+// HasTimeExpr returns true if the expression has a time term.
+func HasTimeExpr(expr Expr) bool {
```

Comment:
Before it was a method on `type SelectStatement` but really it should be a function that works on an `Expr`.  Some of the meta-queries now use it also.